simphony_mayavi/cuds/vtk_lattice.py

```diff
@@ -53,29 +64,35 @@ def __init__(self, name, type_, data_set, data=None):
         self.point_data = CubaData(
             data, stored_cuba=self.supported_cuba, size=size)
 
-        # Estimate the lattice parameters
-        if self.type in ('Cubic', 'OrthorombicP', 'Square', 'Rectangular'):
+        # Determine origin
+        try:
+            self._origin = self.data_set.origin
+        except AttributeError:
```

Comment:
it might be nice to add a comment about why you can have those two cases (or what kind of data structure you have for data_set in which case).